### DISCUSSION TITLE
UFO-R11 - Universal smart IR remote control - handling learned codes

### DISCUSSION THREAD

**Discussion by ivdimitro** [Koenkk/zigbee2mqtt#16281] [Devices]
Hello, I've got the IR remote with the purpose of steering it via MQTT. The issue is that all of the learned codes are concatenated in a long string without obvious separators. Example "Dp8NsgbJAZ4ByQEhBckBniABQAUAySABAJ4gAUAFAMlgAUAHQAMCyQGeIAFABQMhBZ4BQAeAAwLJAZ4gAUAFAMkgAQCeIAFABQDJIAGAJwLJAZ4gAUAFAMkgAQCeIAFABYADAckB4Asj4AMTgAsBngHgAQMByQGAEwHJAUAHAf"

Reading all the codes in some kind of map will be tedious at best. 
Do you know if there is an easy way to extract the learned keys? I haven't connected the device via the official app. Is there a way to export a mapping from there?

**Comment by ophirrom** [comment]
Hi , 
Have you found solution to learn the IR codes ?
Having the same issues

**Reply by ivdimitro** [reply]
I can learn the codes no problem - enable learning --> shoot the remote and the new code is added.  The codes are added in the string given above. In this string there are at least 5 keys. 
The issue is how to map a remote in efficient way. If I have to diff the string after adding an a new code against the old version it will take forever.

**Reply by leah-potato** [reply]
It's an encrypted string. Check out [this write up](https://gist.github.com/mildsunrise/1d576669b63a260d2cff35fda63ec0b5) about a different IR blaster.
